Updated UseGuard documentation

ivorgri opened the pull request:
As mentioned in https://github.com/Savory/Danet/issues/115, the UseGuard documentation was not completely correct. I have also added working examples of a SimpleGuard and JwtGuard to be used by people. Please let me know if you would like to see some changes.

Sorikairox commented:
@ivorgri Thank you very much for your contribution!

Instead of putting the JWT example in the AuthGuard `Overview` part, what do you think of creating a "Recipes" category and putting it inside? Like Nest ?

![Screenshot 2025-06-18 at 23 48 59](https://github.com/user-attachments/assets/585a61c6-6a0e-4366-bd8a-e12be28d930c)

We could then put a link to it in the `Overview` page.

Also, I would personally say first that the example is from Deno website, rather than after the code block, but that's no big deal!

ivorgri commented:
Yeah, I love that idea of doing recipes. Once I have some time, I'll update this pull request with a proposal and tag you once I've done it so you can verify it. Afterwards, I also wanted to change some of the language as it felt a bit "unprofessional"/all over the place. I'll take some time to clean it up and makes some changes.

ivorgri commented:
Hey @Sorikairox , found the time to get this done. I've updated the guards page to include a link to the recipe of the JWT Guard, and put everything inside the Recipes subheader (hope I created those correctly). Also moved the Zod page a bit higher, so there is some more space to put potential other subheaders in between. And as a last small change, updated some broken links on the "Execution context" page. 

Looking forward to feedback!
